docs/middleware/custom.md

```diff
@@ -43,3 +42,16 @@ later, response. Some keys are:
 :body   - the response body
 :response_headers
 ```
+
+### Faraday::Middleware
+
+There's an easier way to write middleware, and it's also the recommended one: make your middleware subclass `Faraday::Middleware`.
+`Faraday::Middleware` already implements the `#call` method for you and looks for two methods in your subclass: `#on_request(env) and `#on_complete(env)`.
+`#on_request` is called when the request is being built and is given the `env` representing the request.
+`#on_complete` is called after the response has been received (that's right, it already supports parallel mode!) and receives the `env` of the response.
+
+### Do I need to override `#call`?
+
+For the majority of middleware, it's not necessary to override the `#call` method, as you can simply use `#on_request` and `#on_response`.
```

Comment:
This on_response should be on_complete, right?

Reply:
YES 🙈

Reply:
Luckily, that was already spot and fixed 🙌 
https://github.com/lostisland/faraday/commit/4e9f4b06f1a1d46fac45663ab3a857aa60b79012

Reply:
Oh nice! I was just reading through the PRs referenced in the release notes. I didn't think to diff the two versions. 😬